.changeset/cuddly-tools-joke.md

```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/itwinui-react": minor
+---
+
+Added optional `labelProps`, `iconProps`, `sublabelProps`, and `checkboxProps` to `TreeNode` component. Added `iconProps` to `TreeNodeExpander`.
```

Comment:
changeset can be updated as follows:

```suggestion
Added optional props to `TreeNode` for customization: `nodeProps`, `contentProps`, `labelProps`, `titleProps`, `sublabelProps`, `checkboxProps`, `iconProps`, `expanderProps` and `subTreeProps`.
```

also PR description can be updated.

Reply:
Committed the suggestion and updated the PR description.